packages/vite/src/lib/entries.ts

```diff
@@ -1,37 +1,24 @@
-import path from 'node:path'
-
-import type { PagesDependency } from '@redwoodjs/project-config'
-import {
-  ensurePosixPath,
-  getPaths,
-  processPagesDir,
-} from '@redwoodjs/project-config'
+import { ensurePosixPath, getPaths } from '@redwoodjs/project-config'
+import { getProject } from '@redwoodjs/structure/dist/index'
+import type { RWPage } from '@redwoodjs/structure/dist/model/RWPage'
+import type { RWRoute } from '@redwoodjs/structure/dist/model/RWRoute'
 
 import { makeFilePath } from '../utils'
 
-const getPathRelativeToSrc = (maybeAbsolutePath: string) => {
-  // If the path is already relative
-  if (!path.isAbsolute(maybeAbsolutePath)) {
-    return maybeAbsolutePath
-  }
-
-  return `./${path.relative(getPaths().web.src, maybeAbsolutePath)}`
-}
-
-const withRelativeImports = (page: PagesDependency) => {
-  return {
-    ...page,
-    relativeImport: ensurePosixPath(getPathRelativeToSrc(page.importPath)),
-  }
-}
-
 export function getEntries() {
   const entries: Record<string, string> = {}
 
+  // Build the entries object based on routes and pages
+  // Given the page's route, we can determine whether or not
+  // the entry requires authentication checks
+  const rwProject = getProject(getPaths().base)
+  const routes = rwProject.getRouter().routes
+
   // Add the various pages
-  const pages = processPagesDir().map(withRelativeImports)
+  const pages = routes.map((route: RWRoute) => route.page) as RWPage[]
+
   for (const page of pages) {
-    entries[page.importName] = page.path
+    entries[page.const_] = ensurePosixPath(page.path)
```

Comment:
Unsure of the use of `const_` is appropriate, but it generates the correct key.

Reply:
What does this do again? Sorry can't remember 😅

Reply:
Use on `const_` here provides the expected key for the entries file that is compatible with the prior way of serving the key:

```
  "UserExampleNewUserExamplePage": "assets/UserExampleNewUserExamplePage-CPG6dVwh.mjs",
  "UserExampleEditUserExamplePage": "assets/UserExampleEditUserExamplePage-BfvIJHxv.mjs",
```

`UserExampleNewUserExamplePage` is the key.